cpmpy/solvers/exact.py

```diff
@@ -0,0 +1,528 @@
+        if not self.objective_minimize:
+            self.objective_value_ = -self.objective_value_
+
+        return True
+
+    def solve(self, time_limit=None, assumptions=None, **kwargs):
+        """
+            Call Exact
+
+            Arguments:
+            - time_limit:  maximum solve time in seconds (float, optional)
+            - assumptions: list of CPMpy Boolean variables (or their negation) that are assumed to be true.
+                           For repeated solving, and/or for use with s.get_core(): if the model is UNSAT,
+                           get_core() returns a small subset of assumption variables that are unsat together.
+
+            Additional keyword arguments:
+            The Exact solver parameters are defined by https://gitlab.com/JoD/exact/-/blob/main/src/Options.hpp#L207
+        """
+        from exact import Exact as xct
+
+        if not self.solver_is_initialized:
+            assert not self.objective_given
+            self.xct_solver.init([],[])
+            self.solver_is_initialized=True
+            self.xct_solver.setOption("verbosity","0")
+        self.xct_solver.clearAssumptions()
+
+        # set additional keyword arguments
+        for (kw, val) in kwargs.items():
+            self.xct_solver.setOption(kw,str(val))
+
+        # set assumptions
+        if assumptions is not None:
+            assert(all(v.is_bool() for v in assumptions))
+            assump_vals = [int(not isinstance(v, NegBoolView)) for v in assumptions]
+            assump_vars = [self.solver_var(v._bv if isinstance(v, NegBoolView) else v) for v in assumptions]
+            self.assumption_dict = {xct_var: (xct_val,cpm_assump) for (xct_var, xct_val, cpm_assump) in zip(assump_vars,assump_vals,assumptions)}
+            for x,v in zip(assump_vars,assump_vals):
+                self.xct_solver.setAssumption(x, [v])
+
+        # call the solver, with parameters
+        start = time.time()
+        my_status = self.xct_solver.runFull(self.objective_given, time_limit if time_limit is not None else 0)
+        end = time.time()
+
+        # new status, translate runtime
+        self.cpm_status = SolverStatus(self.name)
+        self.cpm_status.runtime = end - start
+
+        # translate exit status
+        if my_status == 0: # found unsatisfiability
+            if self.objective_given and self.xct_solver.hasSolution():
+                self.cpm_status.exitstatus = ExitStatus.OPTIMAL
+            else:
+                self.cpm_status.exitstatus = ExitStatus.UNSATISFIABLE
+        elif my_status == 1: # found solution, but not optimality proven
+            assert self.xct_solver.hasSolution()
+            self.cpm_status.exitstatus = ExitStatus.FEASIBLE
+        elif my_status == 2: # found inconsistency over assumptions
+            self.cpm_status.exitstatus = ExitStatus.UNSATISFIABLE
+        elif my_status == 3: # found timeout
+            self.cpm_status.exitstatus = ExitStatus.UNKNOWN
+            return 0
```

Comment:
Should return `False`

Reply:
But ideally, use `self._solve_return(self.status)` which will ensure consistency between solvers as it is a function in the superclass `SolverInterface`

Reply:
See: https://github.com/CPMpy/cpmpy/blob/d53809aa4eabea0808d2842975dfb876e1bdf14d/cpmpy/solvers/ortools.py#L187

Reply:
done